plugins/c8y_log_plugin/src/main.rs

```diff
@@ -0,0 +1,321 @@
+const AFTER_HELP_TEXT: &str = r#"On start, `c8y_log_plugin` notifies the cloud tenant of the log types listed in the `CONFIG_FILE`, sending this list with a `118` on `c8y/s/us`.
+`c8y_log_plugin` subscribes then to `c8y/s/ds` listening for logfile operation requests (`522`) notifying the Cumulocity tenant of their progress (messages `501`, `502` and `503`).
+
+The thin-edge `CONFIG_DIR` is used to store:
+  * c8y-log-plugin.toml - the configuration file that specifies which logs to be retrived"#;
+
+#[derive(Debug, clap::Parser, Clone)]
+#[clap(
+name = clap::crate_name!(),
+version = clap::crate_version!(),
+about = clap::crate_description!(),
+after_help = AFTER_HELP_TEXT
+)]
+pub struct LogfileRequestPluginOpt {
+    /// Turn-on the debug log level.
+    ///
+    /// If off only reports ERROR, WARN, and INFO
+    /// If on also reports DEBUG and TRACE
+    #[clap(long)]
+    pub debug: bool,
+
+    /// Create supported operation files
+    #[clap(short, long)]
+    pub init: bool,
+
+    #[clap(long = "config-dir", default_value = DEFAULT_TEDGE_CONFIG_PATH)]
+    pub config_dir: PathBuf,
+}
+
+async fn create_mqtt_client(
+    tedge_config: &TEdgeConfig,
+) -> Result<mqtt_channel::Connection, anyhow::Error> {
+    let mqtt_port = tedge_config.query(MqttPortSetting)?.into();
+    let mqtt_config = mqtt_channel::Config::default()
+        .with_port(mqtt_port)
+        .with_subscriptions(mqtt_channel::TopicFilter::new_unchecked(
+            C8yTopic::SmartRestRequest.as_str(),
+        ));
+
+    let mqtt_client = mqtt_channel::Connection::new(&mqtt_config).await?;
+    Ok(mqtt_client)
+}
+
+pub async fn create_http_client(
+    tedge_config: &TEdgeConfig,
+) -> Result<JwtAuthHttpProxy, anyhow::Error> {
+    let mut http_proxy = JwtAuthHttpProxy::try_new(tedge_config).await?;
+    let () = http_proxy.init().await?;
+    Ok(http_proxy)
+}
+
+async fn run(
+    config_file: &PathBuf,
+    mqtt_client: &mut Connection,
+    http_client: &mut JwtAuthHttpProxy,
+) -> Result<(), anyhow::Error> {
+    let notify = Arc::new(Notify::new());
+    let notify_async_clone = notify.clone();
+    let config_file_async_clone = config_file.clone();
+
+    let plugin_config = Arc::new(PluginConfig::new(&config_file));
+    let msg = plugin_config.to_supported_config_types_message()?;
+    let () = mqtt_client.published.send(msg).await?;
+
+    tokio::task::spawn(async move {
+        let mut inotify = Inotify::init().expect("Error while initializing inotify instance");
+
+        // Watch for modify and close events.
+        inotify
+            .add_watch(
+                &config_file_async_clone,
+                WatchMask::MODIFY | WatchMask::CLOSE,
+            )
+            .expect("Failed to add file watch");
+
+        // Read events that were added with `add_watch` above.
+        let mut buffer = [0; 1024];
+
+        loop {
+            let events = inotify
+                .read_events_blocking(&mut buffer)
```

Comment:
This blocks the current thread so this may block your event loop for async events...

Reply:
Oh thank you, I will change this to use [read_events](https://docs.rs/inotify/latest/inotify/struct.Inotify.html#method.read_events) instead. It this good enough? @makr11st

Reply:
Sure, that should be fine, alternative would be to use the [event stream](https://docs.rs/inotify/latest/inotify/struct.Inotify.html#method.event_stream) API.